src/sg_client.rs

```diff
@@ -66,16 +68,39 @@ fn make_post_body(mut mail_info: Mail) -> SendgridResult<String> {
 
 impl SGClient {
     /// Makes a new SendGrid cient with the specified API key.
-    pub fn new(key: String) -> SGClient {
+    pub fn new<S: Into<String>>(key: S) -> SGClient {
+        #[cfg(feature = "async")]
+        use reqwest as rq;
+        #[cfg(not(feature = "async"))]
+        use reqwest::blocking as rq;
+
+        let builder = rq::ClientBuilder::new();
+        #[cfg(feature = "rustls")]
+        builder.use_rustls_tls();
+        let client = builder.build().unwrap(); 
+
         SGClient {
-            api_key: key,
-            client: reqwest::blocking::Client::new(),
+            api_key: key.into(),
+            client,
         }
     }
 
-    /// Sends a messages through the SendGrid API. It takes a Mail struct as an
-    /// argument. It returns the string response from the API as JSON.
-    /// It sets the Content-Type to be application/x-www-form-urlencoded.
+    /// Sends a messages through the SendGrid API. It takes a Mail struct as an argument. It returns
+    /// the string response from the API as JSON.
+    /// 
+    /// ### Example
+    /// ```rust
```

Comment:
Since Rust will run documentation tests, this will fail. We may want to add an `ignore` keyword here.